stdlib/mime/src/main/java/org/ballerinalang/mime/util/MimeUtil.java

```diff
@@ -485,4 +486,21 @@ public static BString getMessageAsString(BValue dataSource) {
 
         return new BString(dataSource.stringValue());
     }
+
+    /**
+     * Check whether a given value should be serialized specifically as a JSON.
+     *
+     * @param value Value to serialize
+     * @param entityRecord Entity record
+     * @return flag indicating whether the given value should be serialized specifically as a JSON
+     */
+    public static boolean generateAsJSON(BValue value, BMap<String, BValue> entityRecord) {
+        if (value instanceof BStreamingJSON) {
```

Comment:
Can't we check the type from the type tag here?

Reply:
We can't. type of BStreamingJSON is also JSON. Hence can't distinguish between a normal JSON and a streaming one.